commandline/src/main/java/com/thoughtworks/go/util/command/CommandLine.java

```diff
@@ -327,7 +327,13 @@ public void runScript(Script script, StreamConsumer buildOutputConsumer,
         ProcessWrapper process;
         int exitCode;
 
-        SafeOutputStreamConsumer streamConsumer = new SafeOutputStreamConsumer(new ProcessOutputStreamConsumer(outputStreamConsumer, errorStreamConsumer));
+        SafeOutputStreamConsumer streamConsumer =
+            new SafeOutputStreamConsumer(
+                new BoundedOutputStreamConsumer(
+                    new ProcessOutputStreamConsumer(outputStreamConsumer, errorStreamConsumer),
+                    new SystemEnvironment().get(SystemEnvironment.CONSOLE_LOG_MAX_LINE_LENGTH)
```

Comment:
I'm not happy about reading from the system environment here. It would be better to pass this along the layers, like the `consoleLogCharset`. Requires a lot of changes though, and I wonder if these console related properties should be wrapped into a single `ConsoleLogProperties` data class then, for easier future adjustments.

Reply:
Actually, it might not be so bad, using the `CommandLine::withXYZ` pattern to only set it on demand. Then we can propagate it along the BuildWork layers for the agent. And in all other cases we deliberately avoid setting a maximum line length (e.g. SCM tasks).

In that case, it would make sense to use the `BoundedOutputStreamConsumer` for all of `CommandLine`'s run methods, so the behavior is consistent when using `withMaxLineLength`.

Reply:
I'm not _overly_ worried about this, especially since the `runScript` method is a specific usage style of `CommandLine`, yet is just one (complicated) style with which you can use it.

Given the logic only has one consumer, arguably that's a signal that the logic should be in the `BaseComandBuilder` or in some wrapper class like `LowTrustScriptRunningCommandLine` or something. Honestly, I haven't bothered to try and get my head around all the various consumers and weird abstractions here. Seems overly complicated as it currently is :-) My point is that this is already pretty messy, and although that's not a reason to make it worse, it's probably not THAT much worse adding this :-)

This code is really rather old (and lifted from maven shared utils, based on the copyright); I suspect it might have been around since the original CruiseControl days 😅

The `consoleLogCharset` is even weirder in some ways as it is actually serialized and sent over as part of the `BuildWork`, as if it could reasonably be varied for one job vs another, rather than something defaulted as part of the agent code, and able to be overridden and propagated. So it kind of needs to be propagated through by necessity, rather than serving a function to do so...

Reply:
Your call if you want to have a think about this and submit another small PR :-)